tools/wptrunner/wptrunner/browsers/firefox.py

```diff
@@ -881,10 +890,52 @@ def check_crash(self, process, test):
             return False
 
 
-class FirefoxWdSpecBrowser(NullBrowser):
-    def __init__(self, logger, leak_check=False, **kwargs):
-        super().__init__(logger, **kwargs)
+class FirefoxWdSpecBrowser(WebDriverBrowser):
+    def __init__(self, logger, binary, webdriver_binary, webdriver_args, debug_info=None,
+                 symbols_path=None, stackwalk_binary=None, stackfix_dir=None,
+                 leak_check=False, asan=False, profile=None,
+                 stylo_threads=1, headless=False, enable_webrender=False, chaos_mode_flags=None,
+                 **kwargs):
+
+        super().__init__(logger, binary, webdriver_binary, webdriver_args)
+        self.binary = binary
+        self.webdriver_binary = webdriver_binary
+        self.profile = profile
+
+        self.stackfix_dir = stackfix_dir
+        self.symbols_path = symbols_path
+        self.stackwalk_binary = stackwalk_binary
+
+        self.asan = asan
         self.leak_check = leak_check
+        self.leak_report_file = None
+
+        self.env = get_environ(self.logger,
+                               binary,
+                               debug_info,
+                               stylo_threads,
+                               headless,
+                               enable_webrender,
+                               chaos_mode_flags)
+        self.env["RUST_BACKTRACE"] = "1"
+        # This doesn't work with wdspec tests
```

Comment:
This comment isn't removed from anywhere in this PR, but it is already in `tools/wptrunner/wptrunner/browsers/firefox.py` on `master`. Will it be in two places after this? (Some mishaps due to the PR sitting around seem likely, so I'm doing some spot checking.)